lib/Driver/ToolChains/Flang.cpp

```diff
@@ -294,28 +294,34 @@ void FlangFrontend::ConstructJob(Compilation &C, const JobAction &JA,
     CommonCmdArgs.push_back("8");
   }
 
-  // -g should produce DWARFv2
-  for (auto Arg : Args.filtered(options::OPT_g_Flag)) {
-    Arg->claim();
-    CommonCmdArgs.push_back("-x");
-    CommonCmdArgs.push_back("120");
-    CommonCmdArgs.push_back("0x200");
-  }
+  // Last argument of -g/-gdwarfX should be taken.
+  Arg *GArg = Args.getLastArg(options::OPT_g_Flag);
+  Arg *GDwarfArg = Args.getLastArg(options::OPT_gdwarf_2,
+                                   options::OPT_gdwarf_3,
+                                   options::OPT_gdwarf_4,
+                                   options::OPT_gdwarf_5);
```

Comment:
What happens for "-gdwarf-2 -g"? Shouldn't the "-g" win there?

Reply:
-g suggests dwarf version explicitly, while -gdwarf-N suggests dwarf version explicitly. Implicit suggestion of version is taken only when no explicit suggestion is present. This behavior is exhibited for gfortran and clang as well.
Ex
------------------------ 
$ gfortran -gdwarf-2 -g my.f90
$ llvm-dwarfdump a.out | grep version
0x00000000: Compile Unit: length = 0x000000a5 version = 0x0002 abbr_offset = 0x0000 addr_size = 0x08 (next unit at 0x000000a9)
$ gfortran  -g my.f90
$ llvm-dwarfdump a.out | grep version
0x00000000: Compile Unit: length = 0x0000009d version = 0x0004 abbr_offset = 0x0000 addr_size = 0x08 (next unit at 0x000000a1)
$ clang -gdwarf-2 -g my.c
$ llvm-dwarfdump a.out | grep version
0x00000000: Compile Unit: length = 0x00000050 version = 0x0002 abbr_offset = 0x0000 addr_size = 0x08 (next unit at 0x00000054)
$ llvm-dwarfdump a.out | grep version
0x00000000: Compile Unit: length = 0x00000047 version = 0x0004 abbr_offset = 0x0000 addr_size = 0x08 (next unit at 0x0000004b)

Reply:
Ok. Thanks for the explanation.